primitives/src/lib.rs

```diff
@@ -7,6 +7,8 @@ use codec::{Decode, Encode};
 use enum_iterator::Sequence;
 #[cfg(feature = "std")]
 use scale_info::TypeInfo;
+#[cfg(feature = "runtime")]
```

Comment:
Drink importing this crate with `runtime` feature feels odd.

```suggestion
#[cfg(all(feature = "runtime", feature = "std"))]
```

Reply:
Why do we need `std`?  These imports are only required for the conversion, and the conversion is only enabled for `runtime` feature.